### DISCUSSION TITLE
"wire:init.visible" - Lazy load component data (with scroll + time offset)

### DISCUSSION THREAD

**Discussion by camya** [livewire/livewire#2511] [3. Feature ideas]
There is often content on a website which is not visible "above the fold" and which is not really needed on the initial page load. This could be lazy loaded just if the user scrolls down the page.  

**wire:init** is a great feature for loading data async when the component is rendered on the page. At the moment, it can't be configured to lazy load the component data, if it becomes visible after the user scrolls down the page.

**Feature request: "Lazy Load Livewire Component Data"**

`wire:init.visible="loadDataAsync"`

Loads data via Ajax, if the Livewire component is visible for the user. (See options and defaults below)

**Option "Scroll position offset"** (Scroll offset in px/em/... to trigger the Ajax loading)

`wire:init.visible.250px="loadDataAsync"`

Start loading if component is 250px below (or above) current scroll position.

Default: 500px

**Option "Time offset + Cancel time"** (This avoids the loading of data if a user just scrolls to fast)

`wire:init.visible.300px.100ms="loadDataAsync"`

Start loading data with Ajax after waiting for 100ms, if component is 250px below (or above) current scroll position. If the component is not visible anymore within 100ms (because the user scrolled to fast), the loading is canceled. If the user scrolls to the position of the component again more slowly, the same prodedure is executed again and the component is rendered if the condition match.

Default: 50ms

**Comment by rakeshappycodes** [comment]
Much needed  feature.

**Comment by ChrisSantiago82** [comment]
Yes, I have run into the same issue today with loading many images.
My solution maybe is not the prettiest but I have used the "wire:poll.visible" combined with a if Statement. That way the poll will be removed after the image is loaded

```
    @if ($loaded === false)
        <img src="/assets/loading.gif" wire:poll.visible="loadImage">
    @else
       <img src="https://{{ $image }}">
    @endif 
```

**Comment by f-liva** [comment]
This is NECESSARY to achieve a real performant Livewire approach. Please evaluate it's introduction
